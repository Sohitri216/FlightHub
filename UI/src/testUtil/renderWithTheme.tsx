import { ThemeProvider as MThemeProvider } from "@mui/material/styles";
import { render } from "@testing-library/react";
import { Theme } from "../theme/Theme";

export const renderWithTheme = (component: JSX.Element) => {
  render(
    <MThemeProvider theme={Theme}>
      <>{component}</>
    </MThemeProvider>
  );
};
