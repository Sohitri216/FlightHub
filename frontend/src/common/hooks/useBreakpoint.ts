import { useTheme } from "@mui/material/styles";
import useMediaQuery from "@mui/material/useMediaQuery";

export const useBreakpoint = (): string => {
  const theme = useTheme();
  let screenSize;
  const md = useMediaQuery(theme.breakpoints.between("sm", "md"));
  const lg = useMediaQuery(theme.breakpoints.up("md"));
  const xlg = useMediaQuery(theme.breakpoints.up("lg"));

  if (md) {
    screenSize = "md";
  } else if (lg) {
    screenSize = "lg";
  } else if (xlg) {
    screenSize = "lg";
  } else {
    screenSize = "";
  }

  return screenSize;
};
