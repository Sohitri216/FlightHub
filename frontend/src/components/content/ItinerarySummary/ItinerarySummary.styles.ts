import styled, { css } from "styled-components";

export const IternaryContainer = styled("div")`
  display: flex;
  flex-direction: row;
  flex: 0 0 100%;
  justify-content: space-between;
`;

export const Section = styled("div")`
  display: flex;
  flex-direction: row;
  gap: 0.25rem;
  align-items: center;
`;

export const ItineraryCode = styled("div")<{ type?: string }>`
  ${(props) =>
    props.type === "airport" &&
    css`
      font-size: 0.725rem;
      font-style: italic;
    `}
`;

export const OfferPrice = styled("div")``;

export const Price = styled("div")``;

export const Offer = styled("div")`
  font-style: italic;
  font-size: 0.675rem;
`;

export const JourneyDetailsWrapper = styled("div")``;
export const JourneyDetails = styled("div")``;

export const JourneyRouteBlock = styled("div")`
  display: flex;
  flex-direction: column;
  align-items: flex-start;
`;
