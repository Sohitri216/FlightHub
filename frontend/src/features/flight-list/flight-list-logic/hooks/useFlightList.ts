/* eslint-disable react-hooks/exhaustive-deps */
import axios from "axios";
import { useEffect, useState } from "react";
import {
  FlightParamProps,
  FlightDetails,
  FlightInventoryList,
  ErrorData,
  FlightInventoryData,
} from "../types/types";
export const useFlightList = ({
  origin,
  dest,
  departureDate,
  arrivalDate,
  service,
}: FlightParamProps): FlightInventoryData => {
  const [loading, setLoading] = useState<boolean>(true);
  const [response, setResponse] = useState<FlightDetails[]>([]);
  const [error, setError] = useState<ErrorData>();

  const baseConfig: Params = {
    baseUrl: "http://localhost:3001/promotions/priceoffers/flights/ond",
    headers: { "Content-type": "application/json" },
    method: "get",
  };

  const fetchFlightList = async () => {
    await axios
      .request({
        ...baseConfig,
        url: baseConfig.baseUrl + `/${origin}/${dest}`,
        params: {
          departureDate: departureDate,
          returnDate: arrivalDate,
          service: service,
        },
      })
      .then((response: FlightInventoryList) => {
        setResponse(response?.data);
      })
      .catch((err: ErrorData) => {
        setError(err);
      })
      .finally(() => setLoading(false));
  };

  useEffect(() => {
    fetchFlightList();
  }, []);

  return { response, error, loading };
};

interface Params {
  baseUrl: string;
  headers: any;
  method: string;
}
