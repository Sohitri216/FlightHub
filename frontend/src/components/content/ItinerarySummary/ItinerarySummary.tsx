import { FlightTakeoff } from "@mui/icons-material";
import { useBreakpoint } from "../../../common/hooks/useBreakpoint";
import { SeatLayout } from "../../template/SeatLayout/SeatLayout";
import * as S from "./ItinerarySummary.styles";

export const ItinerarySummary = ({
  depatureDetails: { iataCode: departureIataCode, airport: departureAirport },
  returnDetails: { iataCode: returnIataCode, airport: returnAirport },
  availableSeatCount,
  price,
  offerType,
}: ItineraryType) => {
  const breakpoint = useBreakpoint();

  const DisplayJourneyRoute = ({ iataCode, airport }: JourneyRoute) => {
    return breakpoint === "lg" || breakpoint === "xlg" ? (
      <S.JourneyRouteBlock>
        <S.ItineraryCode>{iataCode}</S.ItineraryCode>
        <S.ItineraryCode type="airport">({airport})</S.ItineraryCode>
      </S.JourneyRouteBlock>
    ) : (
      <S.Section>
        <S.ItineraryCode>{iataCode}</S.ItineraryCode>
      </S.Section>
    );
  };

  return (
    <S.IternaryContainer>
      <S.Section>
        <FlightTakeoff />
        <DisplayJourneyRoute
          iataCode={departureIataCode}
          airport={departureAirport}
        />
      </S.Section>
      <S.Section>
        <FlightTakeoff style={{ transform: "scale(-1,1" }} />
        <DisplayJourneyRoute
          iataCode={returnIataCode}
          airport={returnAirport}
        />
      </S.Section>

      {(breakpoint === "lg" || breakpoint === "xlg") && (
        <S.Section>
          <SeatLayout availableSeatCount={availableSeatCount} />
        </S.Section>
      )}

      <S.OfferPrice>
        <S.Price>{price}</S.Price>
        <S.Offer>{offerType} offer</S.Offer>
      </S.OfferPrice>
    </S.IternaryContainer>
  );
};
type JourneyDetais = {
  country: string;
  airport: string;
  iataCode: string;
  date: string;
};

type ItineraryType = {
  depatureDetails: JourneyDetais;
  returnDetails: JourneyDetais;
  price: string;
  offerType: string;
  availableSeatCount: string;
};

type JourneyRoute = {
  airport: string;
  iataCode: string;
};
