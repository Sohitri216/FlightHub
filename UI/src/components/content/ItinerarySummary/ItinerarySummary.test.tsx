import { renderWithTheme } from "../../../testUtil/renderWithTheme";
import { ItinerarySummary } from "./ItinerarySummary";
import { screen } from "@testing-library/react";
import { useBreakpoint } from "../../../common/hooks/useBreakpoint";

const testProps = {
  depatureDetails: {
    country: "Germany",
    airport: "Frankfurt International Airport",
    iataCode: "FRA",
    date: "15-08-2023",
  },
  returnDetails: {
    country: "Faroe Islands",
    airport: "Vagar Airport",
    iataCode: "FAE",
    date: "30-08-2023",
  },
  offerType: "BestPrice",
  price: "â‚¬200.15",
  availableSeatCount: "31",
};

jest.mock("../../template/SeatLayout/SeatLayout", () => {
  return {
    SeatLayout: () => (
      <div>
        <div data-testid="seat-layout-template">Seat Arranagement</div>
      </div>
    ),
  };
});

jest.mock("../../../common/hooks/useBreakpoint");

describe("<ItinerarySummary /> component", () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it("match its snapshot", () => {
    const view = renderWithTheme(<ItinerarySummary {...testProps} />);
    expect(view).toMatchSnapshot();
  });

  it("should display flight take off icon and flight landing icon", () => {
    renderWithTheme(<ItinerarySummary {...testProps} />);
    expect(screen.getByTestId("takeoff-icon")).toBeDefined();
  });
  it("should display offer price", () => {
    renderWithTheme(<ItinerarySummary {...testProps} />);
    expect(screen.getByTestId("offer-price")).toBeDefined();
  });

  it("should display location name and location code for screen size larger than medium", () => {
    (useBreakpoint as jest.Mock).mockReturnValueOnce("lg");
    renderWithTheme(<ItinerarySummary {...testProps} />);
    expect(screen.getAllByTestId("journey-location")).toBeDefined();
    expect(screen.getAllByTestId("landing-icon")).toBeDefined();
  });

  it("should display seat Layout for greater than small screen", () => {
    (useBreakpoint as jest.Mock).mockReturnValueOnce("lg");
    renderWithTheme(<ItinerarySummary {...testProps} />);
    expect(screen.getByTestId("seat-layout-template")).toBeDefined();
  });

  it("should not render seat Layout for greater than mediumscreen", () => {
    (useBreakpoint as jest.Mock).mockReturnValueOnce("sm");
    renderWithTheme(<ItinerarySummary {...testProps} />);
    expect(
      screen.queryByTestId("seat-layout-template")
    ).not.toBeInTheDocument();
  });
});
