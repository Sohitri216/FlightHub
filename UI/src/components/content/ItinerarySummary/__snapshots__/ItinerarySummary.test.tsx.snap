// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItinerarySummary /> component match its snapshot 1`] = `undefined`;
